Sources/CardPayments/APIRequests/ConfirmPaymentSourceRequest.swift

```diff
@@ -23,8 +23,19 @@ struct ConfirmPaymentSourceRequest: APIRequest {
         var confirmPaymentSource = ConfirmPaymentSource()
         var card = cardRequest.card
         let verification = Verification(method: cardRequest.sca.rawValue)
-        card.attributes = Attributes(verification: verification)
+        
+        if cardRequest.shouldVault {
+            var customer: Customer?
+            let vault = Vault(storeInVault: .onSuccess)
+            if let customerID = cardRequest.customerID {
+                customer = Customer(id: customerID)
+            }
             
+            card.attributes = Attributes(customer: customer, vault: vault, verification: verification)
+        } else {
+            card.attributes = Attributes(customer: nil, vault: nil, verification: verification)
+        }
```

Comment:
I think this could be simplified a bit - see a suggestion I made in this [branch diff](https://github.com/paypal/iOS-SDK/compare/vaulting_with_purchase...vault-wo-purchase-suggestion). Let me know what you think! Feel free to merge that branch into yours directly - the tests will likely need to be updated accordingly.

Reply:
Addressed in commit 6057b4e6155b57c8c2fa4f361c0410eb0f29f1e3 Modified it a bit.